go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -1605,6 +1605,48 @@ func (s *VtctldServer) GetTablets(ctx context.Context, req *vtctldatapb.GetTable
 	}, nil
 }
 
+// GetTopology is part of the vtctlservicepb.VtctldServer interface. It the topology map of the vtctld's topology server.
+func (s *VtctldServer) GetTopology(ctx context.Context, req *vtctldatapb.GetTopologyRequest) (*vtctldatapb.GetTopologyResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "VtctldServer.GetTopology")
+	defer span.Finish()
+
+	resp := vtctldatapb.GetTopologyResponse{}
+
+	cells, err := s.ts.GetKnownCells(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	cells = append([]string{topo.GlobalCell}, cells...)
+	resp.Cells = make([]*vtctldatapb.TopologyCell, len(cells))
+
+	var (
+		wg  sync.WaitGroup
+		rec concurrency.AllErrorRecorder
+	)
+
+	wg.Add(len(cells))
+	for i, cell := range cells {
+		go func(cell string, index int) {
+			defer wg.Done()
+			topoCell, err := s.getTopologyCell(ctx, "/"+cell)
+			if err != nil {
+				rec.RecordError(fmt.Errorf("Error fetching root cell %s: %v", cell, err))
+				return
+			}
+
+			resp.Cells[index] = topoCell
```

Comment:
I'm not sure, this will never read or write to the same place in memory. Is that incorrect?

Reply:
while it's true you have different keys here, go maps have their own internal structure that makes them unsafe for this (second paragraph of [`sync.Map`'s docs](https://pkg.go.dev/sync#Map) -- to be clear i'm _not_ suggesting to use a sync.Map, but a regular map)

here's a toy example: https://go.dev/play/p/2vvSF2JOdnp

Reply:
In this case `Cells` is a slice. I'm familiar with how to use a mutex, thanks. I can use a channel or something similar, but would like to keep it a slice.

Reply:
............ 🤦 i'm an idiot -- i thought we were keying on cell name, i have no idea how i missed this was a slice. super sorry about that!!

(i didn't mean to suggest you didn't know how to use a mutex, the reason i added that last bit was because i couldn't find a good doc on native maps and threadsafety that wasn't on the `sync.Map` page, and i was worried that linking would imply i was pushing for switch to a sync.Map, and i didn't want to cause more confusion. sorry about that too)